### PR TITLE
Add async and await to controllers and sql queries

### PR DESCRIPTION
**Pull request by arnenator**

### CODE REVIEW

File: Minitwit.Infrastructure.Tests/Repositories/FollowerRepositoryTests.cs
Comment (Pinkvinus) [inline Minitwit.Infrastructure.Tests/Repositories/FollowerRepositoryTests.cs:1]: The fact that infrastructure tests are not run before the merge makes me hesitant to approve. Can you please run the tests and show they pass?

**Review by Pinkvinus — CHANGES_REQUESTED** [review]

File: Minitwit.Infrastructure.Tests/Repositories/FollowerRepositoryTests.cs
Comment (arnenator) [reply Minitwit.Infrastructure.Tests/Repositories/FollowerRepositoryTests.cs:1]: I run `dotnet test` in the root of the working directory every time. And it works. It is not up to this PR to add workflows for testing. I will submit a seperate PR that adds this.

Comment (arnenator) [reply Minitwit.Infrastructure.Tests/Repositories/FollowerRepositoryTests.cs:1]: @Pinkvinus I took the liberty to do some changes to the workflow as well. You do not have to merge it. 
If you think what i have done with the workflow is wrong, then let us have a discussion about on friday.

#190

Comment (Pinkvinus) [reply Minitwit.Infrastructure.Tests/Repositories/FollowerRepositoryTests.cs:1]: I never said it was up to this PR to fix this, which is exactly why i asked whether you could show that it worked. I don't think everyone remembers to run dotnet test before they make a PR. It's great that you did, but i don't think my concerns regarding the tests were any less valid, when I am not presented with any evidence of the tests passing.

**Review by Pinkvinus — APPROVED** [review]
Your autoformating software will be the death of me

**Pull request merged**